evcouplings/complex/protocol.py

```diff
@@ -8,15 +8,159 @@
+    median_num_per_species_1 : median number of paralogs per species in the 
+        first monomer alignmment
+    median_num_per_species_2 : median number of paralogs per species in 
+        the second monomer alignment
+    num_with_embl_cds_1 : number of IDs for which we found an EMBL CDS in the 
+        first monomer alignment (relevant to distance concatention only)
+    num_with_embl_cds_2 : number of IDs for which we found an EMBL CDS in the 
+        first monomer alignment (relevant to distance concatention only)
+    
+    Parameters
+    ----------
+    annotation_file_1 : str
+        Path to annotation.csv file for first monomer alignment
+    annotation_file_2 : str
+        Path to annotation.csv file for second monomer alignment
+    genome_location_filename_1 : str
+        Path to genome location mapping file for first alignment
+    genome_location_filename_2 : str
+        PPath to genome location mapping file for second alignment
+    outfile: str
+        Path to output file
+    """
+
+    # load the annotations for each alignment
+    # as {id:species}
+    annotations_1 = read_species_annotation_table(
+        annotation_file_1
+    )
+    species_1 = list(annotations_1.species)
+
+    annotations_2 = read_species_annotation_table(
+        annotation_file_2
+    )
+    species_2 = list(annotations_2.species)
+    
+    # calculate the number of sequences found in each alignment
+    num_seqs_1 = len(annotations_1)
+    num_seqs_2 = len(annotations_2)
+    
+    # calculate the number of species found in each alignment
+    # where a species is defined as a unique OS or Tax annotation field
+    nonredundant_annotations_1 = len(set(species_1))
+    nonredundant_annotations_2 = len(set(species_2))
+
+    # calculate the number of overlapping species
+    species_overlap = list(
+        set(species_1).intersection(set(species_2))
+    )
+    n_species_overlap = len(species_overlap)
+    
+    # calculate the median number of paralogs per species
+    n_paralogs_1 = float(
+        # counts the number of times each species occurs in the list
+        # then takes the median
+        np.median(list(Counter(species_1).values()))
+    )
+
+    n_paralogs_2 = float(
+        np.median(list(Counter(species_2).values()))
+    )
+    
+    # If the user provided genome location files, calculate the number
+    # of ids for which we found an embl CDS
+    if genome_location_filename_1 is not None and \
+        genome_location_filename_2 is not None:
+
+        genome_location_table_1 = pd.read_csv(genome_location_filename_1)
+        genome_location_table_2 = pd.read_csv(genome_location_filename_2)
+
+        # Number uniprot IDs with EMBL CDS that is not NA
+        embl_cds1 = len(list(set(genome_location_table_1.uniprot_ac)))
+        embl_cds2 = len(list(set(genome_location_table_2.uniprot_ac)))
+
+    else:
+        embl_cds1 = None
+        embl_cds2 = None
+
+    concatenation_data = [
```

Comment:
(this is fine, just as a btw: in situations like these I prefer the dict-constructor for dataframes:

```
concatenation_data = [{
    "num_seqs_1": num_seqs_1,
    ...
}]
```
)